docs/Makefile

```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    =
+SPHINXOPTS    = -W # treat warnings as errors
```

Comment:
Possibly a step too far. This will fail the build if we introduce markup that results in warnings in the future.

Reply:
I lean towards not liking this, but also our documentation has been lacking recently so maybe this should be a thing.

Reply:
I also have mixed feelings, but the sheer number of warnings that had crept in would suggest we should maybe try enabling this, and if it turns out to be too annoying, we can always remove it.